packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -1458,6 +1530,9 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
       onEndReachedThreshold,
       initialScrollIndex,
     } = this.props;
+    if (this.state.maintainVisibleContentPositionAdjustment != null) {
+      return;
+    }
```

Comment:
This is to avoid triggering incorrect onStartReachedEvents since scroll position is currently invalid. I will double check, but if I remember correctly without this check right after adding new data at the start of the list `onStartReached` would be called, before we get an adjusted scroll position from native, which shouldn't trigger the callback.

In general whenever `maintainVisibleContentPositionAdjustment` is not null, any function which uses scroll position should not be called. This seems to be safe functions that uses scroll position are usually called again whenever we receive the updated scroll position and reset `maintainVisibleContentPositionAdjustment`.